requires_conditional/conanfile.py

```diff
@@ -15,10 +15,7 @@ def requirements(self):
 
     def build(self):
         cmake = CMake(self)
-        if self.options.zip:
-            cmake.definitions["WITH_ZIP"] = "1"
-        else:
-            cmake.definitions["WITH_ZIP"] = "0"
+        cmake.definitions["WITH_ZIP"] = bool('zlib' in self.deps_cpp_info.deps)
```

Comment:
The problem with this code is that it is not as intuitive for students, that might now now about ``self.deps_cpp_info.deps``. There is probably not enough time to explain everything.

Reply:
I think I would leave ``cmake.definitions["WITH_ZIP"] = bool(self.options.zip)`` for educational purpose, leaving this approach as a post exercise quick comment in a slide. Wdyt?

Reply:
The problem I see with the current source is that when I run  (first step in the corresponding exercise):

```
conan create . user/testing -o hello:zip=False
```

the output is 

```
hello/0.1@user/testing (test package): Running test()
Hello world!
```

It looks like it is not using ZLIB.

----

With my changes the output will be:

```
hello/0.1@user/testing (test package): Running test()
Hello world!
Original size is: 52
Original string is: Conan Package Manager is GREEEEEEEEEEEEEEAAAAAAAAAAT
Compressed size is: 34
Compressed string is: xÚsÎÏKÌHLÎNLOUðMÌRE
™Å
îA®(ÀB
ZLIB VERSION: 1.2.11
```

You see it is using ZLIB even though you are using `-o hello:zip=False`, now it makes sense to modify the recipe to honor the recipe option

Reply:
I think that the line is almost self-explanatory:

```
cmake.definitions["WITH_ZIP"] = bool('zlib' in self.deps_cpp_info.deps)
```

Casting to `bool` the expression `'zlib' in self.deps_cpp_info.deps` (almost as it is read), not sure if it deserves a comment or explanation. I've also added a message to CMake in the line: `message("Add WITH_ZIP definition")`

Reply:
That is the behavior I wanted when ``hello:zip=False``. If that option is False, then the output shouldn't be using zlib at all. That part of the code is correct. Then the job is to remove it as a dependency, because it is not being used.

Reply:
IMO it is confusing: why do I need to remove the dependency if my binary is not using it? I know it is an exercise, but better if fixing that detail in the `conanfile.py` has consequences in the C++ source and the generated binary.... they are more like a C++ developer than like a _graph model engineer_.

Reply:
They already know the previous syntax from the other training exercises (some of them).

Then, you need to remove it, exactly because you are not using it. This is the reason we want to remove it, because our code and our build is not using it in that case. It is not the other way round: hey, let's skip this feature, because I don't have this dependency.

Reply:
ok, np. I'll change only the `cmake.definitions["WITH_ZIP"] = bool(self.options.zip)`